pkgs/unified_analytics/lib/src/utils.dart

```diff
@@ -81,6 +83,78 @@ Directory getHomeDirectory(FileSystem fs) {
   return fs.directory(home!);
 }
 
+/// Returns `true` if user has opted out of legacy analytics in Dart or Flutter
+///
+/// Checks legacy opt-out status for the Flutter
+/// and Dart in the following locations
+///
+/// Dart: `$HOME/.dart/dartdev.json`
+///
+/// Flutter: `$HOME/.flutter`
+bool legacyOptOut({
+  required FileSystem fs,
+  required Directory home,
+}) {
+  final File dartLegacyConfigFile =
+      fs.file(p.join(home.path, '.dart', 'dartdev.json'));
+  final File flutterLegacyConfigFile = fs.file(p.join(home.path, '.flutter'));
+
+  // Example of what the file looks like for dart
+  //
+  // {
+  //   "firstRun": false,
+  //   "enabled": false,  <-- THIS USER HAS OPTED OUT
+  //   "disclosureShown": true,
+  //   "clientId": "52710e60-7c70-4335-b3a4-9d922630f12a"
+  // }
+  try {
+    if (dartLegacyConfigFile.existsSync()) {
```

Comment:
optional: consider extracting the existence checks out of the try blocks to minimize their scope

Reply:
I agree. Also, I would recommend enabling the `avoid_catches_without_on_clause` lint, like we do in the flutter tool: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/analysis_options.yaml#L5

I think catching everything is an anti-pattern (the exception would be when you catch everything, add some logging or send an analytics event, and then rethrow), as it will silently swallow critical bugs we would want to know about, such as runtime type errors.

Reply:
Here it seems like you want to catch `FormatException`

Reply:
Ah I knew there was a lint somewhere that was supposed to catch that, thanks for the suggestion, next push will include the lint + editing the try block scope

Reply:
I actually think the try/catch everything block is a better idea here.  I/O errors are possible if the file can not be read due to permission restrictions and that will result in a different error being thrown.

Reply:
@christopherfujino 's point about catching everything being an anti-pattern in the more general case is correct though, and the try/catch everything should include as little as possible.

Reply:
hmm this lint is actually causing errors now with another area I used a try catch all in the http client

```dart
  /// Receive the payload in Map form and parse
  /// into JSON to send to GA
  Future<http.Response> sendData(Map<String, Object?> body) async {
    try {
      return await _client.post(
        Uri.parse(postUrl),
        headers: <String, String>{
          'Content-Type': 'application/json; charset=UTF-8',
        },
        body: jsonEncode(body),
      );
    } catch (error) {
      return Future<http.Response>.value(http.Response(error.toString(), 500));
    }
  }
```

Thoughts on this block? I didn't want any client to error out if this package was unable to send requests to GA which will allow it to fail silently. Does this need to be editted as well or can i use `// ignore: avoid_catches_without_on_clauses` in this case?

Reply:
In the above example, each client using this package could check the awaited response and look for a `500` error and read the error returned if we wanted check that each event is being sent

Reply:
> I actually think the try/catch everything block is a better idea here. I/O errors are possible if the file can not be read due to permission restrictions and that will result in a different error being thrown.

Good point about permission restrictions.